apps/web/src/utils/page-visibility.ts

```diff
@@ -17,36 +17,38 @@ You should have received a copy of the GNU General Public License
 along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */
 
-var hidden, visibilityChange;
-if (typeof document.hidden !== "undefined") {
+let hidden: string, visibilityChange: string;
+if ("hidden" in document) {
   // Opera 12.10 and Firefox 18 and later support
   hidden = "hidden";
   visibilityChange = "visibilitychange";
-} else if (typeof document.msHidden !== "undefined") {
+} else if ("msHidden" in document) {
   hidden = "msHidden";
   visibilityChange = "msvisibilitychange";
-} else if (typeof document.webkitHidden !== "undefined") {
+} else if ("webkitHidden" in document) {
   hidden = "webkitHidden";
   visibilityChange = "webkitvisibilitychange";
 }
 
-export function onPageVisibilityChanged(handler) {
+export function onPageVisibilityChanged(
+  handler: (status: string, bool: boolean) => void
```

Comment:
Use TypeScript union for specific `status` types here. Currently we have `online`, `offline` & `visibilitychange`

Reply:
Ok